contracts/facade/facets/RevenueFacet.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: BlueOak-1.0.0
+pragma solidity 0.8.19;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../../interfaces/IAssetRegistry.sol";
+import "../../interfaces/IBackingManager.sol";
+import "../../interfaces/IBasketHandler.sol";
+import "../../interfaces/IRToken.sol";
+import "../lib/FacetLib.sol";
+
+/**
+ * @title AuctionsFacet
+ * @notice Single-function facet to return all revenues accumulating across RTokens
+ * @custom:static-call - Use ethers callStatic() to get result after update; do not execute
+ */
+// slither-disable-start
+contract RevenueFacet {
+    struct Revenue {
+        IRToken rToken;
+        IERC20 erc20;
+        uint256 surplus;
+    }
+
+    // normally we don't let facets use storage, but this function requires it
+    // we don't yet have a full solution for sharding storage across facets
```

Comment:
We do, see EIP 7201. (This is what I was talking about when I said I wanted to use namespaced storage in V2 and DAO originally)